src/UID2.Client/DecryptionResponse.cs

```diff
@@ -1,36 +1,30 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
 
 namespace UID2.Client
 {
-    public struct DecryptionResponse
+    public readonly struct DecryptionResponse
     {
-        private readonly DecryptionStatus _status;
-        private readonly string _uid;
-        private readonly DateTime? _established;
-        private readonly int? _siteId;
-        private readonly int? _siteKeySiteId;
-
-        public DecryptionResponse(DecryptionStatus status, string uid, DateTime? established, int? siteId, int? siteKeySiteId)
+        public DecryptionResponse(DecryptionStatus status, string uid, DateTime? established, int? siteId, int? siteKeySiteId, bool? isCstgDerived = false)
         {
-            _status = status;
-            _uid = uid;
-            _established = established;
-            _siteId = siteId;
-            _siteKeySiteId = siteKeySiteId;
+            Status = status;
+            Uid = uid;
+            Established = established;
+            SiteId = siteId;
+            SiteKeySiteId = siteKeySiteId;
+            IsCstgDerived = isCstgDerived;
```

Comment:
best not to use the acronym and call it IsTokenClientSideGenerated as this is something DSPs need to understand so best to be verbose ?

Reply:
Will change